### ISSUE TITLE
Flux clip encoded text won't connect to `positive_text_opt`

### ISSUE THREAD

**Issue by thoraxe** [audioscavenger/save-image-extended-comfyui#35]
I'm trying to connect my flux CLIP text to the positive text option field, but it's not connectable, which makes me think that the conditioning output text of CLIP is not usable with save image extended.

This is also likely some form of user error.

**Comment by Targren** [comment]
Might not be. Connecting my "assembled prompt" (strings fed through a series of "CR text concatenate" nodes, basically) which output "STRING" don't seem to be storing  the positive or negative prompts, despite being connected to the inputs. So the issue may be the node.

**Comment by audioscavenger** [comment]
Some nodes have weird ways of storing or sending data, and some even use JS instead of python, and therefore I am unable to grab the actual values other then a reference name. I believe rgthree's nodes are a good example.
If you could post a PNG or the workflow you have trouble with I could have a look

**Comment by Targren** [comment]
The full workflow is convoluted AF, but I was able to reproduce it by tweaking the simple default comfy workflow and prompt, which should be much easier to deal with, I think. 

I think OP might have been trying to connect the CLIP "conditioning" output directly like some other save-with-metadata nodes use, instead of the raw text. 

![ComfyUI-euler-8 0-20-2024-12-01 16-19-15-0001](https://github.com/user-attachments/assets/626f3b48-2ca0-4f80-8ad4-ffbf72296680)

**Comment by audioscavenger** [comment]
i tried your workflow and all seems to work perfectly
going back to initial issue, what is **flux CLIP text** ?

**Comment by Targren** [comment]
> i tried your workflow and all seems to work perfectly going back to initial issue, what is **flux CLIP text** ?

Maybe I'm not getting what those inputs are supposed to do then? Looking at the metadata, the prompts aren't available anywhere except in the workflow where you can dig them out of the CLIP node?

**Comment by audioscavenger** [comment]
bro I have no clue what you mean what the issue is or what is the bug you are experiencing or if it even a question you have. Please rephrase or use GPT to translate because I really can't understand what you mean.

**Comment by Targren** [comment]
> bro I have no clue what you mean what the issue is or what is the bug you are experiencing or if it even a question you have. Please rephrase or use GPT to translate because I really can't understand what you mean.

TL;DR - When you connect a text node into "positive_prompt_opt" and "negative_prompt_opt," what does the save-image-extended node do with those values?

**Comment by audioscavenger** [comment]
GOOD QUESTION! I had no idea, bu looking at the code:
this will save the **positive_prompt** and **negative_prompt** entries in the JSON file created when you enable **save_job_data**:
![example](https://github.com/audioscavenger/save-image-extended-comfyui/raw/main/assets/save-image-extended-comfyui-jobs-example.png)

**Comment by Targren** [comment]
Okay. I was expecting it to save it in the image metadata as well, not just the sidecar file (which I'm not using). So my expectation was wrong. 

Thanks.
